kubernetes-cluster-kubeone/terraform/run.sh

```diff
@@ -37,20 +37,61 @@ else
 fi
 
 # Extract and check ssh_public_key_file value
-ssh_key_file=$(grep -E "^ssh_public_key_file" terraform.tfvars | awk -F= '{gsub(/[ \047"]/, "", $2); print $2}' | sed 's/^[ \t]*//;s/[ \t]*$//')
+ssh_public_key_file=$(grep -E "^ssh_public_key_file" terraform.tfvars | awk -F= '{gsub(/[ \047"]/, "", $2); print $2}' | sed 's/^[ \t]*//;s/[ \t]*$//')
 
-ssh_key_file="${ssh_key_file/#\~/$HOME}"
+ssh_public_key_file="${ssh_public_key_file/#\~/$HOME}"
 
 
-if ! [ -f "$ssh_key_file" ]; then
-    echo "Error: SSH public key file '$ssh_key_file' not found."
+if ! [ -f "$ssh_public_key_file" ]; then
+    echo "Error: SSH public key file '$ssh_public_key_file' not found."
     echo "Please make sure the file exists and has appropriate permissions"
     exit 1
 else
-    echo -e "Using ssh: \033[32m$ssh_key_file\033[0m"
+    echo -e "Using ssh: \033[32m$ssh_public_key_file\033[0m"
     echo -e "== Please make sure that this key is unique in your Hetzner Cloud account === \n"
 fi
 
+# Extract and check ssh_private_key_file value
+ssh_private_key_file=$(grep -E "^ssh_private_key_file" terraform.tfvars | awk -F= '{gsub(/[ \047"]/, "", $2); print $2}' | sed 's/^[ \t]*//;s/[ \t]*$//')
+
+ssh_private_key_file="${ssh_private_key_file/#\~/$HOME}"
+
+# Check whether an ssh-agent is listening
+if [ -z "$SSH_AUTH_SOCK" ]; then
+    echo "ssh-agent isn't running."
+
+    # Start ssh-agent
+    # NOTE/TODO: The agent sicks around, but only listens to the bash instance launched by the script.
```

Comment:
What about using a bash trap to kill the ssh agent when the script completes or is terminated?